### PR TITLE
Created Workflow for Build and Test

### LINKED ISSUE #16: Setup Github Actions/CI
We need to setup CI for this repo to run tests and generate build (skip this for now)

### PR DESCRIPTION
**Pull request by uzaxirr**
Fixes: #16

### CODE REVIEW

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version = 1
+
+[[analyzers]]
+name = "go"
+enabled = true
+
+  [analyzers.meta]
+  import_root = "github.com/uzaxirr/zestream-server"
```
Comment (abskrj) [inline .deepsource.toml]: will it be zestream or uzaxirr in the url?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+
+    - name: Build
+      run: go build -v
+
+    - name: Test
+      run: cd controllers && go test
```
Comment (abskrj) [inline .github/workflows/go.yml]: i think we should test all the tests in service, utils, controllers...

**Review by abskrj — CHANGES_REQUESTED** [review]

File: .deepsource.toml (hunk 1 above)
Comment (uzaxirr) [reply .deepsource.toml]: This file got added up by mistake, will revert this

File: .github/workflows/go.yml (hunk 2 above)
Comment (uzaxirr) [reply .github/workflows/go.yml]: When i run `go test` in root i get

Screenshot 2023-01-02 at 9 08 43 PM

Any idea what modifications are require to run all the unit test present in the project by firing this cmd from the root

Comment (abskrj) [reply .github/workflows/go.yml]: check this 
https://stackoverflow.com/a/16353449

Comment (uzaxirr) [reply .github/workflows/go.yml]: Thanks!!

**Review by abskrj — APPROVED** [review]
lgtm

**Pull request merged**